src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -173,12 +174,31 @@ typesOf bindingGroupType moduleName vals = withFreshSubstitution $ do
     constrain cs ty = foldr srcConstrainedType ty (map (\(_, _, x) -> x) cs)
 
     -- Apply the substitution that was returned from runUnify to both types and (type-annotated) values
-    tidyUp ts sub = first (map (second (first (second (overTypes (substituteType sub) *** substituteType sub))))) ts
+    tidyUp ts sub = first (map (second (first (second (updateExpr sub *** substituteType sub))))) ts
 
     isHoleError :: ErrorMessage -> Bool
     isHoleError (ErrorMessage _ HoleInferredType{}) = True
     isHoleError _ = False
 
+    updateContextsInExpr :: Substitution -> Expr -> Expr
+    updateContextsInExpr subst = go
+      where
+      (_, go, _) = everywhereOnValues id g id
+      g (TypeClassDictionary con ctx hints) =
+          TypeClassDictionary con (update ctx) hints
+      g expr = expr
+      getLocal = fromMaybe M.empty . M.lookup localScope
+      localScope = Nothing
+      modifyDict f dict = dict { tcdInstanceTypes = f (tcdInstanceTypes dict) }
+      modifyLocal = M.map . M.map . fmap $ modifyDict $ fmap $ substituteType subst
+      update ctx = M.insert localScope (modifyLocal . getLocal $ ctx) ctx
```

Comment:
Just one thought though, could this be done with `M.alter` instead to avoid the need for `getLocal`, etc.?

Reply:
Yup. I should have done a better job looking through Map's api first.

Any thoughts about the following? Could it be clearer and/or more succinct? Also, I'm not sure I have a good sense of the code base's styling conventions. Any customary linting standards I should adopt?
```purescript
    updateContextsInExpr :: Substitution -> Expr -> Expr
    updateContextsInExpr subst = go
      where
      (_, go, _) = everywhereOnValues id g id
      g (TypeClassDictionary con ctx hints) =
          TypeClassDictionary con (updateCtx ctx) hints
      g expr = expr
      updateDict f dict = dict { tcdInstanceTypes = f (tcdInstanceTypes dict) }
      updateScope = fmap . fmap . fmap . fmap $ updateDict $ fmap $ substituteType subst
      updateCtx = M.alter updateScope Nothing
```

Reply:
That revision looks good to me! 

The style/conventions of the codebase are a bit all over the place. As long as there's nothing overtly weird it's all good pretty much :wink:...

Reply:
Cool. Danke beaucoup. I'll submit the changes by tonight.